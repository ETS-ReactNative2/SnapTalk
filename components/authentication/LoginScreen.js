import React from 'react';
import { View, Text, TextInput, TouchableOpacity } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import * as firebase from 'firebase';
import styles from '../authentication/style/LoginScreenStyle';

export default class LoginScreen extends React.Component {

  state = {
    email: '',
    password: '',
    errorMessage: null,
  };

  handleLogin = () => {
    const { email, password } = this.state;

    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .catch(error => this.setState({ errorMessage: error.message }));
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.greeting}>{'Welcome to SnapTalk'}</Text>
        <View style={styles.errorMessage}>
          {this.state.errorMessage && (
            <Text style={styles.error}>{this.state.errorMessage}</Text>
          )}
        </View>

        <KeyboardAwareScrollView
          resetScrollToCoords={{ x: 0, y: 0 }}
          scrollEnabled={true}>
          <View style={styles.form}>
            <View>
              <Text style={styles.inputTitle}>Email Address</Text>
              <TextInput
                style={styles.input}
                autoCapitalize="none"
                onChangeText={email => this.setState({ email })}
                value={this.state.email}></TextInput>
            </View>

            <View style={styles.passwordBox}>
              <View>
                <Text style={styles.inputTitle}>Password</Text>
                <TextInput
                  style={styles.input}
                  secureTextEntry
                  autoCapitalize="none"
                  onChangeText={password => this.setState({ password })}
                  value={this.state.password}></TextInput>
              </View>
            </View>
          </View>

          <TouchableOpacity style={styles.button} onPress={this.handleLogin}>
            <Text style={styles.signIn}>Sign In</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={styles.register}
            onPress={() => this.props.navigation.navigate('Register')}>
            <Text style={styles.newToSnapTalk}>
              New to SnapTalk? <Text style={styles.signUp}>Sign Up</Text>
            </Text>
          </TouchableOpacity>
        </KeyboardAwareScrollView>
      </View>
    );
  }

}