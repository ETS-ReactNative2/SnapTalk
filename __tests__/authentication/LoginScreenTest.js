import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../../components/authentication/LoginScreen'

describe('LoginScreen Test', () => {
    let loginScreenSnapshot = renderer.create(<LoginScreen />).toJSON();

    test('LoginScreen Render Test', () => {
        expect(loginScreenSnapshot).toMatchSnapshot()
    })

    //Authentication is verified by Firebase.
    //Emails and Passwords are verified by Firebase.
    //Therefore, there is nothing else to test for the login page.
})
