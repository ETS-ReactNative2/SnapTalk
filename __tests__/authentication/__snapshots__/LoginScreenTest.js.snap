// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Test LoginScreen Render Test 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../assets/welcome_image.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 120,
            "marginTop": 50,
            "width": 120,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 72,
            "justifyContent": "center",
            "marginHorizontal": 30,
          }
        }
      />
      <View
        style={
          Object {
            "marginBottom": 48,
            "marginHorizontal": 30,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#8A8F9E",
                "fontSize": 10,
                "textTransform": "uppercase",
              }
            }
          >
            Email Address
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "borderBottomColor": "#8A8F9E",
                "borderBottomWidth": 0.5,
                "color": "#161F3D",
                "fontSize": 15,
                "height": 40,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "marginTop": 32,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161F3D",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "purple",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "marginHorizontal": 30,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontWeight": "500",
            }
          }
        >
          Sign In
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "marginTop": 32,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#414959",
              "fontSize": 13,
            }
          }
        >
          New to SnapTalk? 
          <Text
            style={
              Object {
                "color": "#59446A",
                "fontWeight": "500",
              }
            }
          >
            Sign Up
          </Text>
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
